app/modules/submissions/models.py

```diff
@@ -594,9 +594,17 @@ def get_absolute_path(self):
 
         return submission_path
 
+    # TODO should this blow away remote repo?  by default?
     def delete(self):
         for asset in self.assets:
             asset.delete()
         db.session.refresh(self)
-        with db.session.begin():
+        with db.session.begin(subtransactions=True):
             db.session.delete(self)
+
+    # stub of DEX-220 ... to be continued
+    def justify_existence(self):
```

Comment:
Yes the name is amusing but I think delete_if_empty would be more meaningful

Reply:
the reasons i didnt name it something that specific is that i am not 100% sure that is the _only_ criteria we want to use to decide delete-ability.  if that makes sense.  @colinwkingen and i were having a talk about _just when do_ we want to blow away the local repo?  (followed immediately by when do we blow away the remote repo?  eerily applicable to todays woes! haha)

i shall take it under advisement to rename this to something.   maybe we push that _all_ off to ticket dex-220 which is reallyl what was supposed to answer all these things i guess.  i just wanted _something_.  :smile: